mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala

```diff
@@ -201,18 +218,20 @@ object MesosClient extends StrictLogging with StrictLoggingFlow {
               throw MesosRedirectException(leader)
             case _ =>
               response.discardEntityBytes()
-              throw new IllegalArgumentException(s"Mesos server error: $response")
+              import scala.concurrent.duration._
+              val error = Await.result(Unmarshal(response.entity).to[String], 1.minutes)
```

Comment:
I'm really concerned about this await.result in the middle of an active stream stage. Blocking insider the Akka dispatcher is explicitly discouraged.

Can we instead change the map to mapAsync(1), and have the subsequent values be either Future.successful (response, …) etc. perhaps we could use the async/await pattern on the `case _`

Reply:
I've forgot to remove it. I will not add it in this PR.